pkg/apis/stork/v1alpha1/snapshotrestore.go

```diff
@@ -29,18 +27,34 @@ type VolumeSnapshotRestoreSpec struct {
 type SnapshotRestoreStatusType string
 
 const (
-	// SnapshotRestoreStatusInitial is the initial state when snapshot restore is initiated
-	SnapshotRestoreStatusInitial SnapshotRestoreStatusType = ""
-	// SnapshotRestoreStatusPending for when restore is in pending state
-	SnapshotRestoreStatusPending SnapshotRestoreStatusType = "Pending"
-	// SnapshotRestoreStatusReady for when restore is done
-	SnapshotRestoreStatusReady SnapshotRestoreStatusType = "Ready"
-	// SnapshotRestoreStatusError for when restore failed
-	SnapshotRestoreStatusError SnapshotRestoreStatusType = "Error"
+	// VolumeSnapshotRestoreStatusInitial is the initial state when snapshot restore is initiated
+	VolumeSnapshotRestoreStatusInitial SnapshotRestoreStatusType = ""
+	// VolumeSnapshotRestoreStatusPending for when restore is in pending state
+	VolumeSnapshotRestoreStatusPending SnapshotRestoreStatusType = "Pending"
+	// VolumeSnapshotRestoreStatusReady for when restore is done
+	VolumeSnapshotRestoreStatusReady SnapshotRestoreStatusType = "Ready"
+	// VolumeSnapshotRestoreStatusError for when restore failed
+	VolumeSnapshotRestoreStatusError SnapshotRestoreStatusType = "Error"
+)
+
+// VolumeSnapshotRestoreStageType is stage of volume in-place restore
+type VolumeSnapshotRestoreStageType string
+
+const (
+	// VolumeSnapshotRestoreStageInitial is the initial stage when snapshot restore is initiated
+	VolumeSnapshotRestoreStageInitial VolumeSnapshotRestoreStageType = ""
+	// VolumeSnapshotRestoreStageRestore for when restore is in restore stage
+	VolumeSnapshotRestoreStageRestore VolumeSnapshotRestoreStageType = "Restore"
+	// VolumeSnapshotRestoreStageReady for when restore is completed
+	VolumeSnapshotRestoreStageReady VolumeSnapshotRestoreStageType = "Ready"
+	// VolumeSnapshotRestoreStageError for when restore failed
+	VolumeSnapshotRestoreStageError VolumeSnapshotRestoreStageType = "Error"
 )
 
 // RestoreStatus of volume
 type RestoreStatus struct {
+	// Stage of volume snapshot restore
```

Comment:
We should add a `Reason` string in `RestoreVolumeInfo`. On success it should have `Volume restored successfully` and on failure it should have the error message.